trin-core/src/portalnet/storage.rs

```diff
@@ -16,6 +18,9 @@ use super::types::{
 };
 use crate::utils::db::get_data_dir;
 
+// Assuming large 15kB content values, 5k elements consume 75MB of data
+const CACHE_NUM_ELEMENTS_MAX: usize = 5000;
```

Comment:
@carver What do you think of calculating this as a percentage of the `PortalStorage`'s storage capacity? Our default storage capacity is currently 100MB, almost doubling that for the cache seems like a lot. Maybe make this 15-25% of storage capacity?

Reply:
Since we don't offer any limit on RAM usage in trin, there's no obvious setting to calculate this limit from. I'm much more hesitant to permanently use up disk space than I am to temporarily use up memory. Plus, 75MB is a worst-case scenario of storing huge 15kb items, I expect most items would be significantly smaller.